beacon_node/beacon_chain/src/block_verification.rs

```diff
@@ -329,6 +323,17 @@ pub enum BlockError {
     InternalError(String),
 }
 
+/// Which specific signature(s) are invalid in a SignedBeaconBlock
+#[derive(Debug)]
+pub enum InvalidSignature {
+    // The outter signature in a SignedBeaconBlock
+    ProposerSignature,
+    // One or more signatures in BeaconBlockBody
+    BlockBodySignatures,
+    // One or more signatures in SignedBeaconBlock
```

Comment:
what kind of signature error would be not ProposerSignature and BlockBodySignature. Won't these 2 variants encompass all cases?!

Reply:
See https://github.com/sigp/lighthouse/pull/6519/files#r1931489262